lib/connectors/kv-memory.js

```diff
@@ -118,6 +122,28 @@ function(modelName, key, ttl, options, callback) {
   process.nextTick(callback);
 };
 
+KeyValueMemoryConnector.prototype.ttl =
+function(modelName, key, options, callback) {
+  this._removeIfExpired(modelName, key);
+
+  var store = this._getStoreForModel(modelName);
+  var error = new Error('Cannot get TTL for unknown key ' + key);
+
+  // key is unknown
+  if (!(key in store))
+    return process.nextTick(function() { callback(error); });
+
+  var ttl = store[key].getTtl();
+  debug('TTL %j %j -> %s', modelName, key, ttl);
+  // item is expired, but not deleted yet
+  if (ttl < 0)
```

Comment:
This should never happen, because you have already called `_removeIfExpired`, right?

I guess this can happen if we run `_removeIfExpired` just a fraction of millisecond before the key expires, and this `if (ttl<0)` right after the expiration, but I think the probability of that is so low, that it's not worth bothering.